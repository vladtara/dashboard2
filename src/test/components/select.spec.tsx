import { fireEvent, render } from "@testing-library/react";
import Select, { IItemProps } from "../../components/select";

const onChange = jest.fn();
const items: Array<IItemProps> = [
  {
    label: "Test 1",
    value: 0,
  },
  {
    label: "Test 2",
    value: 1,
  },
];

describe("select.tsx", () => {
  it("Tests Select rendering", () => {
    const { baseElement } = render(
      <Select items={items} onChange={(item) => onChange(item)}></Select>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests rendering of multiple Select elements", () => {
    const { baseElement } = render(
      <>
        <Select
          testId="1"
          items={items}
          onChange={(item) => onChange(item)}
        ></Select>
        <Select
          testId="2"
          items={items}
          onChange={(item) => onChange(item)}
        ></Select>
      </>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests Select onChange", () => {
    const select = render(
      <Select items={items} onChange={(item) => onChange(item)}></Select>,
    );

    fireEvent.change(select.getByTestId("select"), { target: { value: 1 } });
    expect(onChange).toBeCalledWith(items[1]);
  });

  it("Tests Select onChange with undefined item", () => {
    const select = render(
      <Select items={items} onChange={(item) => onChange(item)}></Select>,
    );

    fireEvent.change(select.getByTestId("select"), { target: { value: 5 } });
    expect(onChange).toBeCalledWith(undefined);
  });

  it("Tests `current`-value without testId", () => {
    const { baseElement } = render(
      <Select
        current={items[1].label}
        items={items}
        onChange={(item) => onChange(item)}
      ></Select>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests `current`-value with testId", () => {
    const { baseElement } = render(
      <Select
        current={items[1].label}
        testId="1"
        items={items}
        onChange={(item) => onChange(item)}
      ></Select>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
