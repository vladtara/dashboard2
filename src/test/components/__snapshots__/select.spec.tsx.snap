// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select.tsx Tests \`current\`-value with testId 1`] = `
<body>
  <div>
    <select
      class="sc-bczRLJ laxZOh"
      data-testid="select-1"
    >
      <option
        data-testid="option-1-0"
        value="0"
      >
        Test 1
      </option>
      <option
        data-testid="option-1-1"
        value="1"
      >
        Test 2
      </option>
    </select>
  </div>
</body>
`;

exports[`select.tsx Tests \`current\`-value without testId 1`] = `
<body>
  <div>
    <select
      class="sc-bczRLJ laxZOh"
      data-testid="select"
    >
      <option
        data-testid="option-0"
        value="0"
      >
        Test 1
      </option>
      <option
        data-testid="option-1"
        value="1"
      >
        Test 2
      </option>
    </select>
  </div>
</body>
`;

exports[`select.tsx Tests Select rendering 1`] = `
<body>
  <div>
    <select
      class="sc-bczRLJ laxZOh"
      data-testid="select"
    >
      <option
        data-testid="option-0"
        value="0"
      >
        Test 1
      </option>
      <option
        data-testid="option-1"
        value="1"
      >
        Test 2
      </option>
    </select>
  </div>
</body>
`;

exports[`select.tsx Tests rendering of multiple Select elements 1`] = `
<body>
  <div>
    <select
      class="sc-bczRLJ laxZOh"
      data-testid="select-1"
    >
      <option
        data-testid="option-1-0"
        value="0"
      >
        Test 1
      </option>
      <option
        data-testid="option-1-1"
        value="1"
      >
        Test 2
      </option>
    </select>
    <select
      class="sc-bczRLJ laxZOh"
      data-testid="select-2"
    >
      <option
        data-testid="option-2-0"
        value="0"
      >
        Test 1
      </option>
      <option
        data-testid="option-2-1"
        value="1"
      >
        Test 2
      </option>
    </select>
  </div>
</body>
`;
